src/cleanvision/issue_managers/duplicate_issue_manager.py

```diff
@@ -89,14 +125,45 @@ def find_issues(
             issue_type: {} for issue_type in to_compute
         }
 
-        for path in tqdm(filepaths):
-            image = Image.open(path)
-            for issue_type in to_compute:
-                hash = self._get_hash(image, self.params[issue_type])
-                if hash in issue_type_hash_mapping[issue_type]:
-                    issue_type_hash_mapping[issue_type][hash].append(path)
-                else:
-                    issue_type_hash_mapping[issue_type][hash] = [path]
+        if n_jobs is None:
+            if psutil_exists:
+                n_jobs = psutil.cpu_count(logical=False)  # physical cores
+            if not n_jobs:
+                # either psutil does not exist
+                # or psutil can return None when physical cores cannot be determined
+                # switch to logical cores
+                n_jobs = multiprocessing.cpu_count()
+
+        if n_jobs == 1:
+            for path in tqdm(filepaths):
+                image = Image.open(path)
+                for issue_type in to_compute:
+                    hash = self._get_hash(image, self.params[issue_type])
+                    if hash in issue_type_hash_mapping[issue_type]:
+                        issue_type_hash_mapping[issue_type][hash].append(path)
+                    else:
+                        issue_type_hash_mapping[issue_type][hash] = [path]
+        else:
+            args = [
+                {"path": path, "to_compute": to_compute, "params": self.params}
+                for path in filepaths
+            ]
+            with multiprocessing.Pool(n_jobs) as p:
+                results = list(
+                    tqdm(
+                        p.imap_unordered(compute_hash, args, chunksize=10),
+                        total=len(filepaths),
+                    )
+                )
+
+            results = sorted(results, key=lambda r: r["path"])
```

Comment:
and then this would be the common code for collecting results from both n_jobs=1 and otherwise